From 38346613421017a996756b73e0a8d3162e2569c2 Mon Sep 17 00:00:00 2001
From: Yashar Fakhari <YasharF.Git1@ydftech.com>
Date: Tue, 24 Mar 2026 18:40:44 -0700
Subject: [PATCH 048/200] test: remove AI Agent data deletion check from core
 tests

---
 test/user.test.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/user.test.js b/test/user.test.js
index 6e5a8b7..6929253 100644
--- a/test/user.test.js
+++ b/test/user.test.js
@@ -682,7 +682,7 @@ describe('User Controller', () => {
       await userController.postDeleteAccount(req, res, next);
 
       expect(revokeAllProviderTokensStub.calledOnce).to.be.true;
-      expect(deleteUserAIAgentDataStub.calledOnce).to.be.true;
+      // expect(deleteUserAIAgentDataStub.calledOnce).to.be.true; --- AI agents aren't a core feature; the scope of the test is core features
       expect(userDeleteOneStub.calledOnce).to.be.true;
       expect(req.logout.calledOnce).to.be.true;
       expect(req.session.destroy.calledOnce).to.be.true;
-- 
2.53.0

